### PR TITLE
feat: retry plain http in secure mode

### LINKED ISSUE #914: Avoid requiring passing --plain-http when specifying --insecure
### What is the version of your ORAS CLI

 1.0.0

### What would you like to be added?

Currently if the users specifies --insecure the call fails if the server is not responding in http. 

```
oras version Version:    1.0.0+Homebrew Go version: go1.20.2 
oras login -u user--insecure remote-registry:5000 
Error: Get "https://remote-registry:5000/v2/": http: server gave HTTP response to HTTPS client
```
Users are expected to pass a --plain http as well in this case and we can follow precedence from docker for this to make the experience better - https://github.com/docker/cli/blob/de0d30ff24fad55df89adcb3c01d10bb1821c8db/cli/registry/client/client.go#L131

### Why is this needed for ORAS?

Improve user experience. 

### Are you willing to submit PRs to contribute to this feature?

- [ ] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR avoid requiring passing `--plain-http` when `--insecure` is specified

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #914 

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [x]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by TerryHowe** [comment]
I'm confused by this because normally a `--insecure` option does not necessarily mean http

**Comment by qweeah** [comment]
> I'm confused by this because normally a `--insecure` option does not necessarily mean http

Feeling the same. But it is a [proven user experience](https://docs.docker.com/registry/insecure/) in docker. Users can provide one less flag `--plain-http` with minor performance cost. Considering the flag `--insecure` is targeting non-prod usage, such cost is acceptable.

**Comment by sajayantony** [comment]
+1 on reducing the number of flags in ORAS CLI. My understanding is that this is a fallback behavior and not that --insecure will make the transport default to http. Is that the case?

File: test/e2e/suite/command/resolve_host.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func ResolveFlags(reg string, host string, flagType resolveType) []string {
 	resolveFlag := "resolve"
 	usernameFlag := "username"
 	passwordFlag := "password"
-	plainHttpFlag := "plain-http"
+	plainHttpFlag := "insecure"
```
Comment (TerryHowe) [inline test/e2e/suite/command/resolve_host.go:51]: Should this var be renamed? A test for plain-http?

Comment (qweeah) [reply test/e2e/suite/command/resolve_host.go:51]: I was meant to test the https->http auto handling via `--insecure`. This PR is pending for the discussion in https://github.com/oras-project/oras/issues/914, feel free to add your comment there.

Comment (TerryHowe) [reply test/e2e/suite/command/resolve_host.go:51]: Right, but the var name is `plainHttpFlag` and the value is `insecure`

**Comment by TerryHowe** [comment]
Should we kill this PR? Just because docker does something does not make it the correct approach in my book.

**Comment by qweeah** [comment]
Let's close it and keep tracking in #914

**Pull request closed without merging**